RGBMatrixEmulator/emulators/canvas.py

```diff
@@ -1,4 +1,24 @@
 from RGBMatrixEmulator.graphics.color import Color
+import numpy as np
+
+def shift_array(arr, x, y, K):
```

Comment:
I see 2 issues with this approach:

1) does not respect aspect ratio
2) fills pixel buffer with black (0,0,0) instead of only copying the image at the offset

Reply:
For item 1:

Tested using pygame adapter on `dev` branch (left) vs this PR (right) with a 4x4 PNG image

| Old | New |
|--------|--------|
| ![aspect-ratio-old](https://github.com/ty-porter/RGBMatrixEmulator/assets/46663285/4c47c934-ce0f-408e-afe9-67667a890f76) | ![aspect-ratio-new](https://github.com/ty-porter/RGBMatrixEmulator/assets/46663285/3c905ef5-8293-479a-ab41-9aa7934070f2)|

Reply:
For item 2:

Does not appear to match `rpi-rgb-led-matrix` implementation -- this fills the pixel buffer with black (0,0,0) instead of overwriting only the pixels in the new image. That approach would break double-buffering

`SetImage` implementation calls `SetPixelsPillow`
https://github.com/hzeller/rpi-rgb-led-matrix/blob/a3eea997a9254b83ab2de97ae80d83588f696387/bindings/python/rgbmatrix/core.pyx#L22

`SetPixelsPillow` copies image into existing pixel buffer:
https://github.com/hzeller/rpi-rgb-led-matrix/blob/a3eea997a9254b83ab2de97ae80d83588f696387/bindings/python/rgbmatrix/core.pyx#L48-L54

---

Should be testable with a white image larger than canvas via `image-scroller.py` which uses double-buffering:

https://github.com/ty-porter/RGBMatrixEmulator/blob/f0608890a4c096520778ce0aa5fd20e89be1cc94/samples/image-scroller.py#L27-L28

Reply:
Fixed